Modules/Scripted/ShapeAnalysisToolBox/DataImporter.py

```diff
@@ -904,20 +1285,425 @@ def importFiles(self, filePaths):
+    for i_row in range(segment_number):
+      self.InputFreeSurferSegmentsTable.cellWidget(i_row,0).children()[1].blockSignals(True)
+      self.InputFreeSurferSegmentsTable.cellWidget(i_row,0).children()[1].setChecked(False)
+      self.InputFreeSurferSegmentsTable.cellWidget(i_row,0).children()[1].blockSignals(False)
+  
+  def resetFreeSurferTab(self):
+    self.logic.freesurfer_wanted_segments=[]
+    self.uncheckFreeSurferTables()
+  
+  def resetCSVTab(self):
+    #reset CSV tab
+    self.InputCSVFileNameLineEdit.text=''
+
+  def resetDirectoryTab(self):
+    #reset directroy tab
+    self.InputFolderNameLineEdit.text=''
+    self.FolderDirectoryButton.blockSignals(True)
+    self.FolderDirectoryButton.directory = '/'
+    self.FolderDirectoryButton.blockSignals(False)
   '''
   GUI Callback functions
   '''
+  #Shape Analysis Structure Generation Callbacks
+  def onShapeAnalysisFolderChanged(self,directoryPath):
+    logging.debug("onShapeAnalysisFolderChanged: {}".format(directoryPath))
+    # Create a list of files from the directory
+    directory = qt.QDir(directoryPath)
+    if not directory.exists():
+      logging.error("Directory {} does not exist.".format(directory))
+      return
+
+    self.shape_analysis_folder = directoryPath
+    self.InputShapeAnalysisFolderNameLineEdit.text = directoryPath
+
+  def onGenerateShapeAnalysisStructure(self):
+    logging.debug("onGenerateShapeAnalysisStructure")
+
+    #check for imported data
+    if len(self.logic.segmentationDict)==0:
+      logging.error("Empty segmentation dictionary, import data before generating the Shape Analysis Structure.")
+      return
+
+    if self.shape_analysis_folder == '':
+      logging.error("No Shape Analysis folder specified")
+      return
+
+    self.logic.generateShapeAnlaysisStructure(self.shape_analysis_folder)
+
+    print('the shape analysis folder located at %s is ready' %(self.shape_analysis_folder))
+
+  #resize tabs
+  def onCurrentTabChanged(self,index):
+    #resize tabs to fit minimal space
+    for i in range(self.ImporterTypeTabWidget.count):
+      if(i!=index):
+        self.ImporterTypeTabWidget.widget(i).setSizePolicy(qt.QSizePolicy.Ignored, qt.QSizePolicy.Ignored);
+    self.ImporterTypeTabWidget.widget(index).setSizePolicy(qt.QSizePolicy.Preferred, qt.QSizePolicy.Preferred);
+    self.ImporterTypeTabWidget.widget(index).resize(self.ImporterTypeTabWidget.widget(index).minimumSizeHint);
+    self.ImporterTypeTabWidget.widget(index).adjustSize();
+
+    #empty import list
+    self.filteredFilePathsList=[]
+    #reset import option of other tabs
+    tab_text=self.ImporterTypeTabWidget.tabText(index)
+    if tab_text=='Import from FreeSurfer':
+      self.logic.setFreeSurferimport(True)
+      self.logic.setExpectedFileType('VolumeFile')
+      try:
+        self.resetDirectoryTab()
+        self.resetCSVTab()
+      except:
+        pass
+
+    elif tab_text=='Import from CSV':
```

Comment:
checking for the text in a tab is not robust, checking for the index or widget name would be better